lib/net.js

```diff
@@ -961,9 +963,7 @@ Socket.prototype.connect = function(...args) {
     initSocketHandle(this);
   }
 
-  if (cb !== null) {
-    this.once('connect', cb);
-  }
+  this.once('connect', cb);
```

Comment:
I'm not a fan of this. It is unnecessary overhead and increases memory usage when no callback is provided.